cmd/ocm/create/cluster/cmd.go

```diff
@@ -180,58 +182,104 @@ func run(cmd *cobra.Command, argv []string) error {
 	// Get the client for the cluster management api
 	cmv1Client := connection.ClustersMgmt().V1()
 
+	var clusterName string
 	// Check and set the cluster name
-	if len(argv) != 1 || argv[0] == "" {
-		return fmt.Errorf("A cluster name must be specified")
+	if len(argv) == 1 && argv[0] != "" {
+		clusterName = argv[0]
+	} else {
+		if args.interactive {
+			prompt := &survey.Input{
+				Message: "cluster name",
+				Help:    clusterNameHelp,
+			}
+			survey.AskOne(prompt, &clusterName, survey.WithValidator(survey.Required))
+		} else {
+			return fmt.Errorf("A cluster name must be specified")
+		}
+	}
+
+	// Validate flags / ask for missing data.
+
+	// Only offer the 2 providers known to support OSD now;
+	// but don't validate if set, to not block `ocm` CLI from creating clusters on future providers.
+	err = arguments.PromptOrCheckOneOf(cmd.Flags(), "provider", osdProviders())
+	if err != nil {
+		return err
+	}
+
+	err = arguments.PromptBool(cmd.Flags(), "ccs")
+	if err != nil {
+		return err
+	}
+	if args.ccs.Enabled {
+		err = arguments.PromptString(cmd.Flags(), "aws-access-key-id")
```

Comment:
I really like these `PromptFoo` functions! Can we have them take an additional user-friendly name? Alternatively have a map for flag->user-friendly inside the `arguments` pkg and defer to Prompt

Reply:
Sure, will do.   I've help back on that because I wanted to pitch alternative stylings where the prompts would look even more like flags:
```
# The cloud provider to create the cluster on
--provider=▁   # Use arrows to move, type to filter
> aws
  gcp
```
or even like the API request yaml:
```
# The cloud provider to create the cluster on
cloud_provider:
  id: ▁   # Use arrows to move, type to filter
    > aws
      gcp
```

I was thinking this is a command-line API client for power users, and it might be worth being a bit less friendly if it teaches you the the lower-level interfaces it offers (?)
[Well, `ocm` doesn't currently take YAML only JSON, but it could.]

However, for --flag style what we really should do (later) is tab completion on flags themselves.
And one can learn the request structure with --debug.
And I've worked on this too much time to invest in implementing either styling...